File: flutter_app/lib/screens/caregiver_help_screen.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import 'package:provider/provider.dart';
import '../providers/app_state.dart';
import '../theme/app_colors.dart';
import '../widgets/care_header.dart';
import '../widgets/care_card.dart';

class CaregiverHelpScreen extends StatefulWidget {
  const CaregiverHelpScreen({super.key});

  @override
  State<CaregiverHelpScreen> createState() => _CaregiverHelpScreenState();
}

class _CaregiverHelpScreenState extends State<CaregiverHelpScreen> {
  final Set<String> _notifiedCaregiverIds = {};

  @override
  Widget build(BuildContext context) {
    final caregivers = context.watch<AppState>().caregivers;

    return Scaffold(
      backgroundColor: AppColors.pageBg,
      body: Column(
        children: [
          CareHeader(title: 'Ask for Help', onBack: () => context.pop()),
          Expanded(
            child: SingleChildScrollView(
              padding: const EdgeInsets.all(24),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  const Text(
                    'Contact your helper',
                    style: TextStyle(
                      fontSize: 28,
                      fontWeight: FontWeight.w900,
                      color: AppColors.heading,
                    ),
                  ),
                  const SizedBox(height: 8),
                  const Text(
                    'Choose someone to notify for help with your care.',
                    style: TextStyle(
                      fontSize: 18,
                      fontWeight: FontWeight.w500,
                      color: AppColors.mutedText,
                    ),
                  ),
                  const SizedBox(height: 24),
                  ...caregivers.map((c) {
                    final isNotified = _notifiedCaregiverIds.contains(c.id);
                    return Padding(
                      padding: const EdgeInsets.only(bottom: 16),
                      child: CareCard(
                        backgroundColor: isNotified
                            ? AppColors.successBg
                            : AppColors.white,
                        borderColor: isNotified
                            ? AppColors.success
                            : AppColors.border,
                        onTap: () {
                          setState(() => _notifiedCaregiverIds.add(c.id));
                          ScaffoldMessenger.of(context).showSnackBar(
                            SnackBar(
                              content: Text(
                                'Notifying ${c.name}',
                                style: const TextStyle(color: AppColors.white),
                              ),
                              backgroundColor: AppColors.primaryAction,
                            ),
                          );
                        },
                        child: Row(
                          children: [
                            Container(
                              width: 56,
                              height: 56,
                              decoration: BoxDecoration(
                                color: AppColors.caregiver,
                                borderRadius: BorderRadius.circular(18),
                              ),
                              child: const Icon(
                                Icons.person,
                                color: AppColors.purpleBg,
                                size: 28,
                              ),
                            ),
                            const SizedBox(width: 16),
                            Expanded(
                              child: Column(
                                crossAxisAlignment: CrossAxisAlignment.start,
                                children: [
                                  Text(
                                    c.name,
                                    style: TextStyle(
                                      fontSize: 18,
                                      fontWeight: FontWeight.w700,
                                      color: isNotified
                                          ? AppColors.success
                                          : AppColors.heading,
                                    ),
                                  ),
                                  Text(
                                    c.relationship,
                                    style: TextStyle(
                                      fontSize: 16,
                                      color: isNotified
                                          ? AppColors.success
                                          : AppColors.mutedText,
                                      fontWeight: FontWeight.w500,
                                    ),
                                  ),
                                ],
                              ),
                            ),
                            if (!isNotified)
                              Container(
                                padding: const EdgeInsets.all(12),
                                decoration: BoxDecoration(
                                  color: AppColors.primaryAction,
                                  borderRadius: BorderRadius.circular(16),
                                ),
                                child: const Icon(
                                  Icons.send,
                                  color: AppColors.white,
                                  size: 20,
                                ),
                              ),
                            if (isNotified) ...[
                              Container(
                                padding: const EdgeInsets.symmetric(
                                  horizontal: 12,
                                  vertical: 10,
                                ),
                                decoration: BoxDecoration(
                                  color: AppColors.success,
                                  borderRadius: BorderRadius.circular(16),
                                ),
                                child: const Text(
                                  'Notified',
                                  style: TextStyle(
                                    color: AppColors.white,
                                    fontSize: 14,
                                    fontWeight: FontWeight.w700,
                                  ),
                                ),
                              ),
                            ],
                          ],
                        ),
                      ),
                    );
                  }),
                  const SizedBox(height: 24),
                ],
              ),
            ),
          ),
        ],
      ),
    );
  }
}

File: react-native-app/src/screens/CaregiverHelpScreen.js
import React, { useState } from 'react';
import { View, Text, TouchableOpacity, ScrollView, StyleSheet, Alert } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { Ionicons } from '@expo/vector-icons';
import CareHeader from '../components/CareHeader';
import CareCard from '../components/CareCard';
import Colors from '../theme/colors';
import { useAppState } from '../context/AppContext';

export default function CaregiverHelpScreen({ navigation }) {
  const { state } = useAppState();
  const [notified, setNotified] = useState(new Set());

  const handleNotify = (id, name) => {
    setNotified(new Set([...notified, id]));
    Alert.alert('Notification Sent', `${name} has been notified.`);
  };

  return (
    <SafeAreaView style={styles.safe}>
      <CareHeader
        title="Ask for Help"
        onBack={() => navigation.goBack()}
        onEmergency={() => navigation.navigate('Emergency')}
      />
      <ScrollView contentContainerStyle={styles.scroll}>
        <Text style={styles.heading}>Your Caregivers</Text>
        <Text style={styles.body}>Tap a caregiver to send them a notification.</Text>

        {state.caregivers.map((cg) => {
          const isNotified = notified.has(cg.id);
          return (
            <View key={cg.id} style={styles.cardWrapper}>
              <CareCard
                borderColor={isNotified ? Colors.success : Colors.border}
                onTap={() => !isNotified && handleNotify(cg.id, cg.name)}
              >
                <View style={styles.row}>
                  <View style={[styles.avatar, { backgroundColor: isNotified ? Colors.successBg : Colors.blueBg }]}>
                    <Ionicons name="person" size={28} color={isNotified ? Colors.success : Colors.primaryAction} />
                  </View>
                  <View style={{ flex: 1 }}>
                    <Text style={styles.name}>{cg.name}</Text>
                    <Text style={styles.relationship}>{cg.relationship}</Text>
                  </View>
                  {isNotified ? (
                    <View style={styles.notifiedBadge}>
                      <Ionicons name="checkmark-circle" size={16} color={Colors.success} />
                      <Text style={styles.notifiedText}>Notified</Text>
                    </View>
                  ) : (
                    <Ionicons name="send" size={24} color={Colors.primaryAction} />
                  )}
                </View>
              </CareCard>
            </View>
          );
        })}
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  safe: { flex: 1, backgroundColor: Colors.pageBg },
  scroll: { padding: 24 },
  heading: { fontSize: 28, fontWeight: '900', color: Colors.heading, marginBottom: 8 },
  body: { fontSize: 18, color: Colors.mutedText, lineHeight: 28, marginBottom: 24 },
  cardWrapper: { marginBottom: 12 },
  row: { flexDirection: 'row', alignItems: 'center' },
  avatar: {
    width: 56, height: 56, borderRadius: 28,
    justifyContent: 'center', alignItems: 'center', marginRight: 16,
  },
  name: { fontSize: 18, fontWeight: '700', color: Colors.heading },
  relationship: { fontSize: 16, color: Colors.mutedText, marginTop: 2 },
  notifiedBadge: { flexDirection: 'row', alignItems: 'center', gap: 4, backgroundColor: Colors.successBg, borderRadius: 10, paddingHorizontal: 10, paddingVertical: 4 },
  notifiedText: { fontSize: 16, fontWeight: '700', color: Colors.success },
});
